components/mailgun/sources/common-webhook.js

```diff
@@ -0,0 +1,97 @@
+const mailgun = require("../mailgun.app");
+
+module.exports = {
+  props: {
+    mailgun,
+    webhookSigningKey: {
+      type: "string",
+      secret: true,
+      label: "Mailgun webhook signing key",
+      description:
+        "Your Mailgun webhook signing key, found [in your Mailgun dashboard](https://app.mailgun.com/app/dashboard), located under Settings on the left-hand nav and then in API Keys look for webhook signing key. Required to compute the authentication signature of events."
+    },
+    http: "$.interface.http",
+    db: "$.service.db",
+  },
+  methods: {
+    verify(signingKey, timestamp, token, signature) {
+      const crypto = require("crypto");
+      const encodedToken = crypto
+        .createHmac("sha256", signingKey)
+        .update(timestamp.concat(token))
+        .digest("hex");
+      return encodedToken === signature;
+    },
+    emitEvent(eventPayload) {
+      const eventTypes = this.getEventType();
+      if (eventTypes.includes(eventPayload.event)) {
+        const meta = this.generateMeta(eventPayload);
+        this.$emit(eventPayload, meta);
+      }
+    },
+  },
+  hooks: {
+    async activate() {
+      const webhookNames = this.getEventName();
+      for (let i = 0; i < webhookNames.length; i++) {
```

Comment:
a `for ... of` loop will save you one line and I personally prefer them in almost all cases where I'm iterating over an array now:

```javascript
for (const webhookName of webhookNames) {
  // remaining code
}

Reply:
This was interesting. I initially tried and I was getting 0-based array indexes instead of the actual values or each element. I went ahead and this time it worked. I don't know why it didn't work when I implemented it initially.